### PR TITLE
Add conversion of quint Set, exists, and forall

### PR DESCRIPTION
**Pull request by shonfeder**
Contributes to #2438

This includes a non-trivial combinator for converting binary binding
operator from quint to binding operator in ApalacheIR.

-------

- [x] Tests added for any new code
- [x] Ran `make fmt-fix` (or had formatting run automatically on all files edited)
- [x] Documentation added for any new functionality
- [x] [Entries added to `./unreleased/`][changelog format] for any new functionality

[changelog format]: https://github.com/informalsystems/apalache/blob/main/CONTRIBUTING.md#how-to-record-a-change

### CODE REVIEW

File: tla-io/src/test/scala/at/forsyte/apalache/io/quint/TestQuintEx.scala [hunk 1]
```diff
@@ -37,23 +44,33 @@ class TestQuintEx extends AnyFunSuite {
...
     val appBar = QuintApp(uid, "bar", List(_42))
     val letBarBeLambdaInAppBar = QuintLet(uid, barDef, appBar)
-
-    // Builtin operator application
-    def app(name: String, args: QuintEx*): QuintApp = QuintApp(uid, name, args)
+    val nIsGreaterThanZero = app("igt", name, _0)
+    // A predicate on ints
+    val intIsGreaterThanZero = QuintLambda(uid, List("n"), "def", nIsGreaterThanZero)
+    val intSet = app("Set", _1, _2, _3)
+    val emptyIntSet = app("Set", _1, _2, _3)
```
Comment (thpani) [inline tla-io/src/test/scala/at/forsyte/apalache/io/quint/TestQuintEx.scala:51]: I couldn't find where this is used.
Would it make sense to include a test for the empty `Set` application?

File: tla-io/src/test/scala/at/forsyte/apalache/io/quint/TestQuintEx.scala [hunk 2]
```diff
@@ -172,4 +189,16 @@ class TestQuintEx extends AnyFunSuite {
   test("can convert builtin iuminus operator application") {
     assert(convert(Q.app("iuminus", Q._42)) == "-42")
   }
+
+  test("can convert builtin Set operator application") {
+    assert(convert(Q.app("Set", Q.s, Q.s, Q.s)) == """{"s", "s", "s"}""")
+  }
+
+  test("can convert builtin exists operator application") {
```
Comment (thpani) [inline tla-io/src/test/scala/at/forsyte/apalache/io/quint/TestQuintEx.scala:197]: Would it make sense to include (negative) tests for the 2 exceptional cases in `binaryBindingApp`?

**Review by thpani — APPROVED** [review]
Hooray! 🎉 

I think we lack a test for empty `Set` application, otherwise LGTM

File: tla-io/src/test/scala/at/forsyte/apalache/io/quint/TestQuintEx.scala (hunk 1 above)
Comment (shonfeder) [reply tla-io/src/test/scala/at/forsyte/apalache/io/quint/TestQuintEx.scala:51]: Yes! Absolutely. That war an oversight. Thanks!

File: tla-io/src/test/scala/at/forsyte/apalache/io/quint/TestQuintEx.scala (hunk 2 above)
Comment (shonfeder) [reply tla-io/src/test/scala/at/forsyte/apalache/io/quint/TestQuintEx.scala:197]: I couldn't decide on those. The logic gets a bit more complicated, but there is still only really one way that constructor can go right, and looking at the code it is pretty evident to me that every other way will go wrong and result in the error... 

But maybe it worth adding a test for that.

**Comment by shonfeder** [comment]
Thanks for the review!

**Review by konnov — APPROVED** [review]
Looks great to me too!
